test/integration/builder/transaction.js

```diff
@@ -659,5 +659,29 @@ module.exports = function(knex) {
         return builder;
       });
     });
+
+    it('#3690 does not swallow exception when error during transaction occurs', async () => {
+      const killConnectionMap = {
+        mysql: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        mysql2: async (connection) => knex.raw('KILL ?', [connection.threadId]),
+        pg: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        redshift: async (connection) =>
+          knex.raw('SELECT pg_terminate_backend(?)', [connection.processID]),
+        mssql: async (connection, trx) =>
+          knex.raw('KILL ?', [(await trx.raw('select @@SPID as id'))[0].id]),
+        oracledb: async (connection) => connection.drop(),
+      };
+
+      const killConnection = killConnectionMap[knex.client.driverName];
+      if (!killConnection) return;
+
+      await expect(
+        knex.transaction(async (trx2) => {
+          await killConnection(await trx2.client.acquireConnection(), trx2);
+          await trx2.transaction(async () => 2);
+        })
```

Comment:
@maximelkin : It looks like some of the exceptions are being thrown for the wrong reason.  Try removing the `await expect(..)` wrapper and watch what happens.  Here's what I'm seeing on my side when I do that:

```

  1) Integration Tests
       oracle | oracledb
         Transactions
           #3690 does not swallow exception when error during transaction occurs:
     TypeError: connection.drop is not a function
      at oracledb (test/integration/builder/transaction.js:674:52)
      at /Users/brianlauber/Documents/src/knex/test/integration/builder/transaction.js:682:15

  2) Integration Tests
       mssql | mssql
         Transactions
           #3690 does not swallow exception when error during transaction occurs:
     RequestError: KILL @p0 - Incorrect syntax near '@p0'.


  3) Integration Tests
       postgresql | pg
         Transactions
           #3690 does not swallow exception when error during transaction occurs:
     Error: ROLLBACK - Client has encountered a connection error and is not queryable

```

(The errors for `oracledb` and `mssql` look incorrect to me)

Reply:
Sorry, not tested locally oracle and mssql

Reply:
Ah -- np.  It took me a while to figure out a good way to test those ones locally.  I'll see if I can figure out a good solution for the Oracle / MSSQL ones

Reply:
mssql works differently with placeholders?

Reply:
ok, its not allowed to kill with placeholder